internal/provider/resource_integer.go

```diff
@@ -17,11 +19,11 @@ func resourceInteger() *schema.Resource {
 			"This resource can be used in conjunction with resources that have the `create_before_destroy` " +
 			"lifecycle flag set, to avoid conflicts with unique names during the brief period where both the " +
 			"old and new resources exist concurrently.",
-		Create: CreateInteger,
-		Read:   schema.Noop,
-		Delete: schema.RemoveFromState,
+		CreateContext: CreateInteger,
+		ReadContext:   schema.NoopContext,
+		DeleteContext: DeleteInteger,
```

Comment:
It's annoying that there is no `schema.RemoveFromStateContext`

Reply:
@detro your suggestion makes sense to me. Have created a shared `DeleteContext` func and added it to `provider.go`.